distributed/shuffle/tests/test_merge.py

```diff
@@ -222,10 +218,8 @@ async def test_merge(c, s, a, b, how, disk):
             await c.compute(dd.merge(A, b, how=how)),
             pd.merge(A, B, how=how),
         )
-        # Note: No await since A and B are both pandas dataframes and this doesn't
-        # actually submit anything
         list_eq(
-            c.compute(dd.merge(A, B, how=how)),
```

Comment:
It looks like this wouldn't work on "legacy" `dask.dataframe` anymore. Is this intended?

Reply:
I've confirmed that this raises with `TypeError: object DataFrame can't be used in 'await' expression`.

Reply:
I think I avoided that on purpose. But I don't have a very strong opinion either way. The thing I didn't like about this is that we return a pandas DataFrame instead of a Dask DataFrame In this case

Reply:
Super, first thought is I like receiving dask dataframes when I use merge from dask, so made it conditional in https://github.com/dask/distributed/pull/8560/commits/a165f1601de7bec2637cd8c4cf458f6f91f0f2fc, but happy to adapt.